modules/programs/antidote.nix

```
@@ -0,0 +1,71 @@
+{ config, lib, pkgs, ... }:
+
+with builtins;
+with lib;
+let
+  module_config = config.programs.zsh.antidote;
+
+  zPluginStr = with lib;
+    (pluginNames:
+      optionalString (pluginNames != [ ]) "${concatStrings (map (name: ''
+        ${name}
+      '') pluginNames)}");
+  trimLines = (text:
+    concatLines (map (p: removePrefix "     " p) (splitString "\n" text)));
+in {
+  options.programs.zsh.antidote = {
+    enable = mkEnableOption "antidote - a zsh plugin manager";
+
+    plugins = mkOption {
+      default = [ ];
+      type = types.listOf types.str;
+      description = "List of antidote plugins, strings.";
+    };
+
+    useFriendlyNames = mkOption {
+      default = false;
+      type = types.bool;
+      description = "use friendlyPath.";
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.antidote;
+      description = "custom antidote pacakge.";
+    };
+
+    antidoteHome = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      defaultText = "~/.antidote or ~/Library/Caches/antidote(darwin).";
+      apply = toString;
+      description = "Path to antidote home directory, cache contents.";
+    };
+  };
+
+  config = mkIf module_config.enable {
+
+    home.packages = [ module_config.package ];
+
+    home.file.".zsh_plugins.txt".text = zPluginStr (module_config.plugins);
```

Maybe this has to be something like
```suggestion
    home.file."${relToDotDir ".zsh_plugins.txt"}".text = zPluginStr (module_config.plugins);
```
? In other words, if the user has configured a special `ZDOTDIR`, will antidote also look there?

---

such a good advice ,  i've change to follow zsh.dotdir option